api/log-handler.go

```diff
@@ -11,10 +12,12 @@ type logHandler struct {
 }
 
 func (l logHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	span, ctx := opentracing.StartSpanFromContext(r.Context(), "logHandler")
+	defer span.Finish()
 	// Serves the request.
 	logger := ContextLogger(r)
 	logger.Info("Start serving", r.Method, r.Host, r.URL)
-	l.handler.ServeHTTP(w, r)
+	l.handler.ServeHTTP(w, r.WithContext(ctx))
```

Comment:
What happens if tracer is disabled? `ctx` equals r.Context() or nil?

Reply:
r.Context()

Reply:
r.Context()